extensions/amp-instagram/1.0/test/test-instagram.js

```diff
@@ -0,0 +1,151 @@
+      <Instagram
+        shortcode="B8QaZW4AQY_"
+        style={{
+          'width': 500,
+          'height': 600,
+        }}
+      />,
+      {attachTo: el}
+    );
+
+    const iframe = wrapper.find('iframe');
+
+    expect(iframe.prop('src')).to.equal(
+      'https://www.instagram.com/p/B8QaZW4AQY_/embed/?cr=1&v=12'
+    );
+    expect(wrapper.find('iframe').prop('style').width).to.equal('100%');
+    expect(wrapper.find('iframe').prop('style').height).to.equal('100%');
+    expect(wrapper.find('div')).to.have.lengthOf(2);
+  });
+
+  it('Render with caption', () => {
+    const el = document.createElement('div');
+    document.body.appendChild(el);
+    const wrapper = mount(
+      <Instagram
+        shortcode="B8QaZW4AQY_"
+        captioned
+        style={{'width': 500, 'height': 705}}
+      />,
+      {attachTo: el}
+    );
+    expect(wrapper.find('iframe').prop('src')).to.equal(
+      'https://www.instagram.com/p/B8QaZW4AQY_/embed/captioned/?cr=1&v=12'
+    );
+    expect(wrapper.find('iframe').prop('style').width).to.equal('100%');
+    expect(wrapper.find('iframe').prop('style').height).to.equal('100%');
+    expect(wrapper.find('div')).to.have.lengthOf(2);
+  });
+
+  it('Resize prop is called', () => {
+    const el = document.createElement('div');
+    document.body.appendChild(el);
+    const requestResizeSpy = env.sandbox.spy();
+    mount(
+      <Instagram
+        shortcode="B8QaZW4AQY_"
+        captioned
+        style={{'width': 500, 'height': 705}}
+        requestResize={requestResizeSpy}
+      />,
+      {attachTo: el}
+    );
+
+    const mockEvent = createMockEvent();
+    mockEvent.source = document.querySelector('iframe').contentWindow;
+    window.dispatchEvent(mockEvent);
+
+    expect(requestResizeSpy).to.have.been.calledOnce;
+  });
+
+  it('Height is changed', async () => {
+    const el = document.createElement('div');
+    document.body.appendChild(el);
+    const wrapper = mount(
+      <Instagram
+        shortcode="B8QaZW4AQY_"
+        style={{'width': 500, 'height': 600}}
+      />,
+      {attachTo: el}
+    );
+
+    const mockEvent = createMockEvent();
+    mockEvent.source = document.querySelector('iframe').contentWindow;
+    window.dispatchEvent(mockEvent);
+
+    wrapper.update();
+
+    await waitForHeight(
+      document.querySelector('iframe').parentElement.parentElement,
+      '1000px'
+    );
+
+    expect(
+      document.querySelector('iframe').parentElement.parentElement.style.height
+    ).to.equal('1000px');
+  });
+});
+
+function createMockEvent() {
+  const mockEvent = new CustomEvent('message', {
+    detail: {
+      origin: 'https://www.instagram.com',
+    },
+  });
+  mockEvent.origin = 'https://www.instagram.com';
+  mockEvent.data = JSON.stringify({
+    'type': 'MEASURE',
+    'details': {
+      'height': 1000,
+    },
+  });
+  return mockEvent;
+}
+
+function waitForHeight(element, height) {
+  return new Promise((resolve, reject) => {
```

Comment:
Use `waitFor` to implement this. Search other tests. It's used a lot.

Reply:
Ah found it, thanks! I was looking for it in enzyme and thought it was maybe from another testing library since it wasn't in there.